src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java

```
@@ -137,11 +161,46 @@ protected String getSObjectQuery(String sObjectName, Schema schema) {
-    if (!containsMacro(propertyName) && propertyValue < SalesforceConstants.INTERVAL_FILTER_MIN_VALUE) {
+  private void validateSObjectDatetimeFilter(String propertyName, String datetime) {
+    if (containsMacro(propertyName) || Strings.isNullOrEmpty(datetime) || datetime.trim().isEmpty()) {
+      return;
+    }
+    try {
+      ZonedDateTime.parse(datetime, DateTimeFormatter.ISO_DATE_TIME);
+    } catch (DateTimeParseException e) {
       throw new InvalidConfigPropertyException(
-        String.format("Invalid SObject '%s' value: '%d'. Value must be '%d' or greater", propertyName, propertyValue,
-          SalesforceConstants.INTERVAL_FILTER_MIN_VALUE), propertyName);
+        String.format("Invalid SObject '%s' value: '%s'. Value must be in Salesforce Date Formats. For example, "
+                        + "2019-01-01T23:01:01Z", propertyName, datetime), propertyName);
+    }
+  }
+
+  private void validateSObjectRangeFilter(String propertyName, Map<ChronoUnit, Integer> rangeValue) {
+    if (containsMacro(propertyName) || rangeValue.isEmpty()) {
+      return;
+    }
+    List<Map.Entry<ChronoUnit, Integer>> invalidValues = rangeValue.entrySet().stream()
+      .filter(e -> e.getValue() < SalesforceConstants.RANGE_FILTER_MIN_VALUE)
+      .collect(Collectors.toList());
+
+    if (!invalidValues.isEmpty()) {
+      throw new InvalidConfigPropertyException(
+        String.format("Invalid SObject '%s' values: '%s'. Values must be '%d' or greater", propertyName,
+                      invalidValues, SalesforceConstants.RANGE_FILTER_MIN_VALUE), propertyName);
+    }
+  }
+
+  private Map<ChronoUnit, Integer> extractRangeValue(String rangeValue) {
+    if (Strings.isNullOrEmpty(rangeValue)) {
+      return Collections.emptyMap();
     }
+    return Stream.of(rangeValue.split(","))
+      .map(String::trim)
+      .map(s -> s.split(":", 2))
+      .filter(keyValue -> keyValue.length > 1)
```

this ignores invalid input, but I think it's better to throw an exception.